include/ginkgo/core/base/std_extensions.hpp

```diff
@@ -64,6 +64,17 @@ struct make_void {
 }  // namespace detail
 
 
+template <typename T>
+struct dont_deduce {
+    using type = T;
+};
+
+
+// Avoid a parameter being involved in type deduction
+template <typename T>
+using dont_deduce_t = typename dont_deduce<T>::type;
```

Comment:
true, I didn't remember the matching name. TBH, this is probably the most unnecessary change, I can do fine without it as well, since this was just for a handful of function calls 😄

Reply:
I removed it, this feels like too much effort for little pay-off :smile: